department-of-veterans-affairs/vets-api
# Add backup form 526 submission path

## Description of change


This change adds an additional sidekiq job, that gets queued when a Form526Submission job is exhausted.
Once it is exhausted (meaning it won't get retried and is dead in the water at this point), this PR adds a new sidekiq job that parses the submission, generates a bunch of PDFs, makes 1 call to EVSS (to get a paper version of the 526 submission), and then sends everything to Central Mail.

## Original issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/50111

## Related PRs
https://github.com/department-of-veterans-affairs/devops/pull/12353
https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/pull/1214
https://github.com/department-of-veterans-affairs/vets-api/pull/11386


## Things to know about this PR

- There is additions to settings.yml,under `form526_backup`
- There is a feature flag `form526_backup_submission_temp_killswitch`, however it is also toggleable via settings. It must be enabled in settings AND feture toggled on, to be "on"
- Currently it submits as 1 payload to central mail, with everything else as an "attachment" to the main payload, which is the 526 PDF
- This can optionally be switched to 1 submission per document (which sets the docType correctly). The VA says they would rather have it 1 payload even if all the docs dont have the correct docType that way (because you can only add 1 docType per payload), however I feel like this cannot be right, or in the future they might rather each do come in with the correct docType set, so it is togglable in the settings `Settings.form526_backup.submission_method = 'single' || 'multi'`



- Testing has been done locally, and test cases added.
- Merge
- Test in staging via Staging Rails console
- Confirm it worked as expected (got 526 paper pdf from EVSS staging, CMP got payload)
- Once that is confirmed, test an actual failed submission in prod rails console
- Confirm with CMP they recived the claim and can action it
- Enable for reals, and monitor closely, turn off via Flipper killswitch if required

TODO:

- [x] add job tracker test cases to make sure it fires this job
